packages/snaps-execution-environments/scripts/build.js

```diff
@@ -0,0 +1,197 @@
+        description: 'Whether to regenerate the LavaMoat policy or not',
+        type: 'boolean',
+      }),
+  );
+
+  await Promise.all(
+    Object.entries(ENTRY_POINTS).map(async ([key, config]) => {
+      console.log('Bundling', key);
+      const { html, entryPoint, node } = config;
+      const insertGlobalVars = node
+        ? { process: undefined, ...LavaMoatBrowserify.args.insertGlobalVars }
+        : LavaMoatBrowserify.args.insertGlobalVars;
+      const bundler = browserify(entryPoint, {
+        ...LavaMoatBrowserify.args,
+        insertGlobalVars,
+        extensions: ['.ts'],
+        node,
+      });
+
+      if (node) {
+        bundler.external([
+          'worker_threads',
+          'buffer',
+          'stream',
+          'tty',
+          'crypto',
+          'util',
+          'events',
+          'os',
+          'timers',
+          'fs',
+        ]);
+      }
+
+      bundler.transform(require('babelify'), {
+        extensions: ['.js', '.ts'],
+        presets: [
+          require('@babel/preset-typescript'),
+          [
+            require('@babel/preset-env'),
+            {
+              targets: {
+                browsers: ['chrome >= 80', 'firefox >= 78'],
+              },
+            },
+          ],
+        ],
+        plugins: [
+          [
+            require('babel-plugin-tsconfig-paths-module-resolver'),
+            {
+              resolvePath: (sourcePath, currentFile, opts) => {
+                const result = defaultResolvePath(
+                  sourcePath,
+                  currentFile,
+                  opts,
+                );
+                // Result is null for modules found in node_modules
+                if (!result) {
+                  return null;
+                }
+                // Force resolve `@metamask/snaps-utils` to browser bundle regardless of environment to reduce bundle size
+                // Use default resolver for everything else
+                if (
+                  sourcePath === '@metamask/snaps-utils' &&
+                  result.includes('../snaps-utils/src')
+                ) {
+                  return `${result}/index.browser`;
+                }
+                return result;
+              },
+            },
+          ],
+        ],
+      });
+
+      // Tree shaking
+      bundler.transform(require('@browserify/uglifyify'), { global: true });
+      bundler.plugin(require('common-shakeify'), { ecmaVersion: 2020 });
+
+      bundler.plugin(LavaMoatBrowserify, {
+        writeAutoPolicy,
+        policy: path.resolve(
+          __dirname,
+          `../lavamoat/browserify/${key}/policy.json`,
+        ),
+        policyName: key,
+        policyOverride: path.resolve(
+          __dirname,
+          `../lavamoat/browserify/policy-override.json`,
+        ),
+        // Prelude is included in Node, in the browser it is inlined.
+        includePrelude: node,
+      });
+
+      const buffer = await new Promise((resolve, reject) => {
```

Comment:
(nitpick) we're in node, so this could be `util.promisify`

Reply:
I tried this, but it seems to not be compatible with Browserify. I decided against spending more time figuring out why.

Reply:
it has nothing to do with browserify, it's node builtin util. Anyway, as long as this is small enough and we can see resolve/reject cannot be called twice it's ok.